index.bs

```diff
@@ -1141,6 +1243,218 @@ steps:
+1. Let |tail| be an empty [=byte sequence=].
+1. If |writePosition| + |data|.[=byte sequence/length=] is smaller than |oldSize|:
+   1. Let |tail| be a [=byte sequence=] containing the last
+      |oldSize| - (|writePosition| + |data|.[=byte sequence/length=]) bytes of |fileContents|.
+1. Let |newSize| be the [=byte sequence/length=] of the concatenation of |head|, |data| and |tail|.
+1. If |newSize| - |oldSize| exceeds the available [=storage quota=], throw a {{QuotaExceededError}}.
+1. Set [=this=].[=[[file]]=]'s [=file entry/binary data=] to the concatenation of |head|, |data| and |tail|.
+1. If the operations modifying the [=this=].[=[[file]]=]'s [=file entry/binary data=] in the previous steps
+   failed:
+    1. If there were partial modifications and the number of modified bytes is known,
+       return the number of modified bytes.
+    1. Else return 0.
+1. Return |data|.[=byte sequence/length=].
+
+</div>
+
+### The {{FileSystemSyncAccessHandle/truncate()}} method ### {#api-filesystemsyncaccesshandle-truncate}
+
+<div class="note domintro">
+  : |handle| . {{FileSystemSyncAccessHandle/truncate()|truncate}}(|newSize|)
+  :: Resizes the file associated with stream to be |newSize| bytes long. If size is larger than the current file size this pads the file with null bytes, otherwise it truncates the file.
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemSyncAccessHandle>truncate(|newSize|)</dfn> method, when invoked, must run
+these steps:
+
+1. If [=this=].[=[[state]]=] is "closed", throw a {{InvalidStateError}}.
+1. If  |newSize| is less than 0, throw a {{InvalidStateError}}.
+1. Let |fileContents| be a copy of [=this=].[=[[file]]=]'s [=file entry/binary data=].
+1. Let |p| be [=a new promise=].
+1. Run the following steps [=in parallel=]:
+  1. Let |oldSize| be the [=byte sequence/length=] of [=this=].[=[[file]]=]'s [=file entry/binary data=].
+  1. If |newSize| is larger than |oldSize|:
+     1. If |newSize| - |oldSize| exceeds the available [=storage quota=], [=/reject=] |p|
+        with a {{QuotaExceededError}} and abort.
+     1. Set [=this=].[=[[file]]=]'s to a [=byte sequence=] formed by concating
+        |fileContents| with a [=byte sequence=]
+        containing |newSize|-|oldSize| `0x00` bytes.
+  1. Else if |newSize| is smaller than |oldSize|:
+     1. Set [=this=].[=[[file]]=]'s to a [=byte sequence=] containing the first |newSize| bytes
+        in |fileContents|.
+  1. [=/Resolve=] |p|.
+1. Return |p|
+
+</div>
+
+### The {{FileSystemSyncAccessHandle/getSize()}} method ### {#api-filesystemsyncaccesshandle-getsize}
+
+<div class="note domintro">
+  : |handle| . {{FileSystemSyncAccessHandle/getSize()}}
+  :: Returns the size of the file associated with |handle| in bytes.
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemSyncAccessHandle>getSize()</dfn> method, when invoked, must run
+these steps:
+
+1. If [=this=].[=[[state]]=] is "closed", throw a {{InvalidStateError}}.
```

Comment:
Similar issue around returning a rejected promise instead of throwing, and exception style.